Library/Homebrew/sandbox.rb

```diff
@@ -89,6 +105,53 @@ def deny_write_homebrew_repository
     end
   end
 
+  def deny_read_shell_files
```

Comment:
I talked about various things further up. I have mixed feelings about the shell configuration files & fairly strong feelings about the history files.

In my `zshrc` because I'm lazy I have functions that flip my active GPG key around depending on whether I'm signing personal stuff or signing code stuff. That's not a super novel idea & there's certainly a chunk of folk who have keys they keep non-public & given Homebrew doesn't police network calls during build it seems kinder to be overzealous on presuming everyone is potentially compromised rather than assuming everyone has good intentions & perfect security. Fundamentally we don't _lose_ anything by protecting those files, so. As long as functionality isn't impeded I think a sandbox potentially too broad is a better option than a sandbox potentially too narrow.

On the history files that is a significantly more important one though, for pretty obvious reasons. Not everything masks passwords on the command line, you can discover the software people use, where they spend the most time working around their system from the CLI, the servers they `ssh` into, the things they `wget`, `curl`, `youtube-dl`, etc off the internet, the `gpg` keys they request and by extension some of the people they potentially communicate with, etc.

It also handles this annoying UX issue every time you debug a build in `zsh`, although I suspect the same applies to most history files given the nature of the error:
```
zsh: locking failed for /Users/xyz/.zsh_history: operation not permitted: reading anyway
```